Nodejs/Product/Nodejs/Project/NodejsProjectNode.cs

```diff
@@ -202,7 +201,11 @@ public NodejsProjectNode(NodejsProjectPackage package)
 
         public bool IsTypeScriptProject {
             get {
-                return string.Equals(GetProjectProperty(NodejsConstants.EnableTypeScript), "true", StringComparison.OrdinalIgnoreCase);
+                var task = ProjectMgr.Site.GetUIThread().InvokeAsync(() => {
+                    return string.Equals(GetProjectProperty(NodejsConstants.EnableTypeScript), "true", StringComparison.OrdinalIgnoreCase);
+                });
+                task.Wait();
+                return task.Result;
```

Comment:
why invoke the UI thread here? If it is indeed necessary to do so, we should probably cache the value so we would only need to make a blocking call once per session.